sklearn/_build_utils/__init__.py

```diff
@@ -33,3 +35,38 @@ def atlas_not_found(blas_info_):
         cblas_libs = blas_info.pop('libraries', [])
 
     return cblas_libs, blas_info
+
+
+def tweak_extensions_to_build_from_c_and_cpp_files(extensions):
+    """Modify the extensions to build from the .c and .cpp files.
+
+    This is useful for releases, this way cython is not required to
+    run python setup.py install.
+    """
+    for extension in extensions:
+        sources = []
+        for sfile in extension.sources:
+            path, ext = os.path.splitext(sfile)
+            if ext in ('.pyx', '.py'):
+                if extension.language == 'c++':
+                    ext = '.cpp'
+                else:
+                    ext = '.c'
+                sfile = path + ext
+            sources.append(sfile)
+        extension.sources = sources
+
+
+def tweak_extensions(top_path, config):
```

Comment:
Can we call this something less opaque: `conditional_cythonize` or `cythonize_if_development`?